### DISCUSSION TITLE
How to use Timers set in Webui in "if and"

### DISCUSSION THREAD

**Discussion by gbananachewicz** [arendst/Tasmota#15723] [Q&A] [answered]
How to use Timers set in Webui in scripting  "if and"?

**Comment by gemu2015** [comment, marked as answer]
don't know what exactly you want to do, but here is an example how to visualize tasmota timers on the webui main page

\>D
res=0
mod=0
timstr=""
tstr=""
cnt=0
rpwr=0
rstr=""
tim=0

\>B

#show
if wm==0
then
res=ttget(0 8)
if res==0 {
wcs {s}Timer ausgeschaltet{e}
} else {
wcs {s}Timer eingeschaltet{e}
}
	
wcs {s}\<hr>{e}
	
for cnt 1 16 1
res=ttget(cnt 7)
if res>0 {
rpwr=ttget(cnt 5)
mod=ttget(cnt 6)
switch mod
case 0
; time
tim=ttget(cnt 0)
tstr="Uhrzeit"
case 1
; sunrise
tim=sunrise
tstr="Sonnenaufgang"
case 2
; sunset
tim=sunset
tstr="Sonnenuntergang"
ends
dp2.0
timstr=s(int(tim/60))+":"+s(tim%60)
		
if rpwr==1 {
rstr="ON"
} else {
rstr="OFF"
}
dp0.0
wcs {s}Timer %0cnt% - %timstr% - %tstr%{m}%rstr%{e}
} else {
;wcs {s}Timer %cnt%{m}OFF{e}
}
next
}

\>W
%=#show

<img width="479" alt="Bildschirmfoto 2022-05-31 um 19 59 31" src="https://user-images.githubusercontent.com/11647075/171253990-caed72a2-f6db-48ad-9990-18b648c777ce.png">

**Comment by gbananachewicz** [comment]
![Screenshot_20220601-082413](https://user-images.githubusercontent.com/25414202/171341000-88d9bce3-bc5e-43ce-b597-8729b29a8ada.png)
I want to use schedule from http://192.168.5.149/tm? In script.
If time1
and pH>10
...
endif


time1-. Is what I don't know - how to get timer enabled with working time

**Reply by gemu2015** [reply]
this is a bit complicated.
like in my example above you must decode wether the timer is in realtime or sunset, sunrise mode.
then you will have the minutes from midnight of the timer.
in the script variable time you have the current time in minutes from midnight with which you can compare if you are in the interval.
however since the tasmota timer normally sets a relay you may simply ask the power status of the relay

if pwr[1]>0
and pH>10
...
